sdk/eventhub/event-hubs/src/eventHubBufferedProducerClient.ts

```diff
@@ -0,0 +1,315 @@
+   * e.g. 'Endpoint=sb://my-servicebus-namespace.servicebus.windows.net/;SharedAccessKeyName=my-SA-name;SharedAccessKey=my-SA-key;'.
+   * @param eventHubName - The name of the specific Event Hub to connect the client to.
+   * @param options - A set of options to apply when configuring the client.
+   * - `retryOptions`   : Configures the retry policy for all the operations on the client.
+   * For example, `{ "maxRetries": 4 }` or `{ "maxRetries": 4, "retryDelayInMs": 30000 }`.
+   * - `webSocketOptions`: Configures the channelling of the AMQP connection over Web Sockets.
+   * - `userAgent`      : A string to append to the built in user agent string that is passed to the service.
+   */
+  constructor(
+    connectionString: string,
+    eventHubName: string,
+    options: EventHubBufferedProducerClientOptions
+  ); // eslint-disable-line @azure/azure-sdk/ts-naming-options
+  /**
+   * The `EventHubBufferedProducerClient` class is used to send events to an Event Hub.
+   * Use the `options` parmeter to configure retry policy or proxy settings.
+   * @param fullyQualifiedNamespace - The full namespace which is likely to be similar to
+   * <yournamespace>.servicebus.windows.net
+   * @param eventHubName - The name of the specific Event Hub to connect the client to.
+   * @param credential - An credential object used by the client to get the token to authenticate the connection
+   * with the Azure Event Hubs service.
+   * See &commat;azure/identity for creating credentials that support AAD auth.
+   * Use the `AzureNamedKeyCredential` from &commat;azure/core-auth if you want to pass in a `SharedAccessKeyName`
+   * and `SharedAccessKey` without using a connection string. These fields map to the `name` and `key` field respectively
+   * in `AzureNamedKeyCredential`.
+   * Use the `AzureSASCredential` from &commat;azure/core-auth if you want to pass in a `SharedAccessSignature`
+   * without using a connection string. This field maps to `signature` in `AzureSASCredential`.
+   * @param options - A set of options to apply when configuring the client.
+   * - `retryOptions`   : Configures the retry policy for all the operations on the client.
+   * For example, `{ "maxRetries": 4 }` or `{ "maxRetries": 4, "retryDelayInMs": 30000 }`.
+   * - `webSocketOptions`: Configures the channelling of the AMQP connection over Web Sockets.
+   * - `userAgent`      : A string to append to the built in user agent string that is passed to the service.
+   */
+  constructor(
+    fullyQualifiedNamespace: string,
+    eventHubName: string,
+    credential: TokenCredential | NamedKeyCredential | SASCredential,
+    options: EventHubBufferedProducerClientOptions // eslint-disable-line @azure/azure-sdk/ts-naming-options
+  );
+  constructor(
+    fullyQualifiedNamespaceOrConnectionString1: string,
+    eventHubNameOrOptions2: string | EventHubBufferedProducerClientOptions,
+    credentialOrOptions3?:
```

Comment:
I am so jealous of union types....

Reply:
We do get the best stuff in TypeScript :D